packages/mail-client/src/mail-client.ts

```diff
@@ -122,7 +177,11 @@ async function sendMailWithNodemailer<T extends MailOptions>(
   mailDestination: MailDestination,
   ...mailOptions: T[]
 ): Promise<MailResponse[]> {
-  const transport = createTransport(mailDestination);
+  let socket: SocksSocket | undefined;
+  if (mailDestination.proxyType === 'OnPremise') {
+    socket = await createSocket(mailDestination);
+  }
+  const transport = await createTransport(mailDestination, socket);
```

Comment:
[pp] I know the `for` loop was already there from the last PR. I prefer the pattern of a await Promise.all(opts.map.(opt,idx=>{})) I assume that is also stronger parallelized, but if it is intentional to not overload the server keep it.

Reply:
I discussed with Marika about the sequential v.s. parallel and we decided to have a small ticket (https://github.com/SAP/cloud-sdk-backlog/issues/820) to make the implementation/test easier.
Once we have the e2e tests in the pipeline, we can send emails in parallel.